compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -464,8 +465,15 @@ object Checking {
       fail(em"only classes can be ${(sym.flags & ClassOnlyFlags).flagsString}")
     if (sym.is(AbsOverride) && !sym.owner.is(Trait))
       fail(AbstractOverrideOnlyInTraits(sym))
-    if (sym.is(Trait) && sym.is(Final))
-      fail(TraitsMayNotBeFinal(sym))
+    if sym.is(Trait) then
+      if sym.is(Final) then
+        fail(TraitsMayNotBeFinal(sym))
+      else if sym.is(Open) then
+        warn(RedundantModifier(Open))
+    if sym.isAllOf(Abstract | Open) then
+      warn(RedundantModifier(Open))
+    if sym.is(Open) && sym.owner.is(Method) then
```

Comment:
This can be generalized to:
```suggestion
    if sym.is(Open) && sym.isLocal then
```
To also include val owners, local class owner, etc.

Reply:
Fixed.